receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -224,5 +423,11 @@ func (r *pReceiver) Shutdown(context.Context) error {
 	if r.unregisterMetrics != nil {
 		r.unregisterMetrics()
 	}
+	if r.apiServer != nil {
+		err := r.apiServer.Shutdown(ctx)
+		if err != nil {
+			r.settings.Logger.Error("API server shutdown failed", zap.Error(err))
```

Comment:
should we return this error instead of logging it?

Reply:
Thanks David yes, made this change and the comment above